experimental/devices/as7262/as7262.go

```diff
@@ -118,8 +125,13 @@ func (d *Dev) Sense(ledDrive physic.ElectricCurrent, senseTime time.Duration) (S
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
-	d.ctx, d.cancel = context.WithCancel(context.Background())
-	defer d.cancel()
+	d.done = make(chan struct{}, 1)
```

Comment:
I think this needs to be rewritten as:
```
done := make(chan struct{}, 1)
d.canceledMu.Lock()
d.canceled = false
d.done = done
d.canceledMu.Unlock()
defer func() {
  done <- struct{}{}
}()
```
and then use done instead of d.done below, as d.done is updated only with d.canceledMu lock but not d.mu in Halt().

Reply:
I've made the changes, and I think its right now.